openforcefield/utils/utils.py

```diff
@@ -274,6 +294,130 @@ def string_to_quantity(quantity_string):
     output_quantity = _ast_eval(ast.parse(quantity_string, mode='eval').body)
     return output_quantity
 
+def convert_smirnoff_data_strings_to_quantity(smirnoff_data):
```

Comment:
I'd simpy call this `convert_all_strings_to_quantity`. This works with any `dict` and `list` if I understand correctly, and doesn't require the data to be in a particular format connected to SMIRNOFF, if I understand this correctly.

Reply:
Agreed. That's clearer. Changed.